tests/test_custom_expert.py

```diff
@@ -0,0 +1,50 @@
+import os
+import pytest
+from typing import Optional
+
+import torch
+
+import hivemind
+from hivemind import RemoteExpert, background_server
+
+@pytest.mark.forked
+def test_custom_expert(port: Optional[int] = None, hid_dim=16):
+    with background_server(
+        expert_cls='perceptron', num_experts=2, device='cpu',
+        hidden_dim=hid_dim, num_handlers=2, no_dht=True,
+        custom_module_path=os.path.join(os.path.dirname(__file__), 'custom_networks.py')) as (server_endpoint, _):
+
+        expert0 = RemoteExpert('expert.0', server_endpoint)
+        expert1 = RemoteExpert('expert.1', server_endpoint)
+
+        for batch_size in (1, 4):
+            batch = torch.randn(batch_size, hid_dim)
+
+            output0 = expert0(batch)
+            output1 = expert1(batch)
+
+            loss = output0.sum()
+            loss.backward()
+            loss = output1.sum()
+            loss.backward()
+
+@pytest.mark.forked
+def test_multihead_expert(port: Optional[int] = None, hid_dim=16):
+    with background_server(
+        expert_cls='multihead', num_experts=2, device='cpu',
+        hidden_dim=hid_dim, num_handlers=2, no_dht=True,
+        custom_module_path=os.path.join(os.path.dirname(__file__), 'custom_networks.py')) as (server_endpoint, _):
+
+        expert0 = RemoteExpert('expert.0', server_endpoint)
+        expert1 = RemoteExpert('expert.1', server_endpoint)
+
+        for batch_size in (1, 4):
```

Comment:
Not sure if it's fixed, still see 3 batch sizes. For checking custom experts even 1 (maybe 2 for the worst case) are sufficient

Reply:
It's a tuple, not a range, which means I check only batch_size equal to 1 and 4.
To sum up, what I attempt to test:
- custom_experts work correctly bot with batch_size equal to 1 and not 1
- different custom_experts do not conflict if they are defined in one file
- custom experts fine well with any kind of input e.g. multiple tensors

Reply:
My two tests cover all of these aspects, but I'm not sure they do so in the most efficient, brief, and fancy way.